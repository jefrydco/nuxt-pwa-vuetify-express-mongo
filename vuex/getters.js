export default {
  isNotif: state => state.notif.text !== null && state.notif.color !== null
};
