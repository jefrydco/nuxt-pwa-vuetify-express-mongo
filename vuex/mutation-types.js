export default {
  SET_NOTIF: "SET_NOTIF"
};
