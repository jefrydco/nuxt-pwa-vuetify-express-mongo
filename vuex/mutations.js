import types from "./mutation-types";

export default {
  [types.SET_NOTIF]: (state, { text, color }) => {
    state.notif = {
      text,
      color
    };
  }
};
