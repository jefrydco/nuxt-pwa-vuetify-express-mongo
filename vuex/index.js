import getters from "./getters";
import mutations from "./mutations";
import actions from "./actions";

const state = () => ({
  notif: {
    text: null,
    color: null
  }
});

export default {
  state,
  getters,
  mutations,
  actions
};
