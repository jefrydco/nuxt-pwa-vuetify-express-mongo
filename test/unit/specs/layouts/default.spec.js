import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Vuetify from "vuetify";
import VeeValidate from "vee-validate";
import moment from "moment";
import axios from "axios";

import Default from "~/layouts/default";

import createRepo from "~/helpers/api";
import vuex from "~/store";

// https://github.com/vuetifyjs/vuetify/issues/4068#issuecomment-446988490
const logError = console.error;
console.error = (...args) => {
  if (
    args[0].includes("[Vuetify]") &&
    args[0].includes("https://github.com/vuetifyjs/vuetify/issues/4068")
  )
    return;
  logError(...args);
};

const localVue = createLocalVue();
const createApi = createRepo(axios);
const apiList = {
  products: createApi("products")
};

localVue.use(Vuex);
localVue.use(Vuetify);
localVue.use(VeeValidate);

describe("Default layout", () => {
  let store;

  beforeAll(() => {
    store = new Vuex.Store({
      ...vuex
    });
  });

  it("renders to match snapshot", () => {
    const wrapper = shallowMount(Default, {
      localVue,
      store,
      mocks: {
        $moment: moment,
        $api: apiList
      },
      stubs: {
        nuxt: "nuxt"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
