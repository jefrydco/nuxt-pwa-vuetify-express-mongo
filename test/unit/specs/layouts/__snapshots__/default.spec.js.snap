// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default layout renders to match snapshot 1`] = `
<v-app-stub id="app">
  <v-content-stub tag="main">
    <v-container-stub tag="div" fluid="" grid-list-xl="">
      <v-layout-stub tag="div" row="" wrap="">
        <v-flex-stub tag="div" xs12="">
          <div></div>
        </v-flex-stub>
      </v-layout-stub>
    </v-container-stub>
  </v-content-stub>
  <app-notif-stub></app-notif-stub>
</v-app-stub>
`;
